csharp/extractor/Semmle.Extraction.CSharp/Entities/Expression.cs

```diff
@@ -593,5 +603,31 @@ public SymbolInfo SymbolInfo
         }
 
         public NullableFlowState FlowState => TypeInfo.Nullability.FlowState;
+
+        public bool IsInsideIfDirective()
+        {
+            return Node.Ancestors().Any(a => a is ElifDirectiveTriviaSyntax || a is IfDirectiveTriviaSyntax);
```

Comment:
I am a bit worried about performance here, as we will traverse the full parents hierarchy, which feels like worst-case quadratic time.

Reply:
Yep, this is not ideal. I've limited the use a bit further, now only `Name.Create(ExpressionNodeInfo info)` uses it, which only calls this method if all else failed, so the expression is either debug symbol or an unknown one.